model_training/get_current_lr.py

```
@@ -0,0 +1,13 @@
+#################################################################
+# --------->>>| Function that always returns current learning
+# --------->>>| rate even for different schedulers
+# ###############################################################
+def current_lr(optim, sched=None):
+    # Varlid for modern schedulers: StepLR, CosineAnnealingLR, etc.
+    if sched is not None and hasattr(sched, "get_last_lr"):
+        try:
+            return sched.get_last_lr()[0]
+        except Exception:
+            pass
```

If this fails the code should quit, not fail silently. I would prefer a single line instead of a try-except block.

---

What if I just added a print statement or something?

---

If we let things fail silently, there is a chance that we get results with hidden problems. I like failures to be loud unless we know we want to brush it under the rug.